core/src/data_sources/data_source.rs

```diff
@@ -800,194 +802,81 @@ impl DataSource {
-                            None => (),
-                        };
-                        match tags.is_not.clone() {
-                            Some(v) => {
-                                must_not_filter.push(qdrant_match_field_condition("tags", v))
-                            }
-                            None => (),
-                        };
-                    }
-                    None => (),
-                };
-
-                match f.parents {
-                    Some(parents) => {
-                        match parents.is_in.clone() {
-                            Some(v) => must_filter.push(qdrant_match_field_condition("parents", v)),
-                            None => (),
-                        };
-                        match parents.is_not.clone() {
-                            Some(v) => {
-                                must_not_filter.push(qdrant_match_field_condition("parents", v))
-                            }
-                            None => (),
-                        };
-                    }
-                    None => (),
-                };
-
-                match f.timestamp {
-                    Some(timestamp) => {
-                        match timestamp.gt.clone() {
-                            Some(v) => must_filter.push(
-                                qdrant::FieldCondition {
-                                    key: "timestamp".to_string(),
-                                    range: Some(qdrant::Range {
-                                        gte: Some(v as f64),
-                                        ..Default::default()
-                                    }),
-                                    ..Default::default()
-                                }
-                                .into(),
-                            ),
-                            None => (),
-                        };
-                        match timestamp.lt.clone() {
-                            Some(v) => must_filter.push(
-                                qdrant::FieldCondition {
-                                    key: "timestamp".to_string(),
-                                    range: Some(qdrant::Range {
-                                        lte: Some(v as f64),
-                                        ..Default::default()
-                                    }),
-                                    ..Default::default()
-                                }
-                                .into(),
-                            ),
-                            None => (),
-                        };
-                    }
-                    None => (),
-                };
-
-                Some(qdrant::Filter {
-                    must: must_filter,
-                    must_not: must_not_filter,
-                    ..Default::default()
-                })
+        let chunks = match query.is_none() {
```

Comment:
generally match query {
  Some(..) =>
  None => 
 }
 
 prevents you from unwraping in the is not none case

Reply:
Fixed